src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +190,19 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        if action_type and action_type == "variable":
+            form_var = FormVariable.objects.get(key=variable)
+
+            if form_var.data_type == "date":
```

Comment:
I think here and on line 193 (`action_type == "variable"`) you can use `FormVariableDataTypes.date` and  `LogicActionTypes.variable` respectively.

This helps in case we ever change the string, then we don't have to change strings everywhere but just in one place.